File: app/src/main/java/com/eure/traveling/DribbbleService.java
package com.eure.traveling;

import com.eure.traveling.entity.Shot;

import java.io.IOException;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Retrofit;
import retrofit2.converter.jackson.JacksonConverterFactory;
import retrofit2.http.GET;
import retrofit2.http.Query;

public final class DribbbleService {
    public static final String API_URL = "https://api.dribbble.com/v1/";

    public interface Dribbble {
        @GET("shots/?access_token=" + BuildConfig.DRIBBBLE_ACCESS_TOKEN)
        Call<List<Shot>> shots(@Query("list") String type, @Query("page") int page);
    }

    public static void main(Callback<List<Shot>> callback, String type, int page) throws IOException {
        Retrofit retrofit = new Retrofit.Builder().baseUrl(API_URL)
                .addConverterFactory(JacksonConverterFactory.create())
                .build();

        Dribbble dribbble = retrofit.create(Dribbble.class);

        Call<List<Shot>> call = dribbble.shots(type, page);

        call.enqueue(callback);
    }
}

File: app/src/main/java/com/eure/traveling/DribbbleService.kt
package com.eure.traveling

import com.eure.traveling.entity.Shot
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Retrofit
import retrofit2.converter.jackson.JacksonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query
import java.io.IOException

object DribbbleService {
    val API_URL = "https://api.dribbble.com/v1/"

    interface Dribbble {
        @GET("shots/?access_token=" + BuildConfig.DRIBBBLE_CLIENT_ACCESS_TOKEN)
        fun shots(@Query("list") type: String, @Query("page") page: Int): Call<List<Shot>>
    }

    @Throws(IOException::class)
    fun main(callback: Callback<List<Shot>>, type: String, page: Int) {
        val retrofit = Retrofit.Builder().baseUrl(API_URL)
                .addConverterFactory(JacksonConverterFactory.create())
                .build()

        val dribbble = retrofit.create<Dribbble>(Dribbble::class.java)

        val call = dribbble.shots(type, page)

        call.enqueue(callback)
    }
}
